src/main/java/org/broadinstitute/hellbender/engine/MultiVariantDataSource.java

```diff
@@ -252,10 +252,15 @@ private VCFHeader getMergedHeader() {
                 .map(ds -> getHeaderWithUpdatedSequenceDictionary(ds))
                 .collect(Collectors.toList());
 
+        final Set<String> allSamples = new LinkedHashSet<>();  //use a set to uniquify samples
+        for (final VCFHeader h : headers) {
```

Comment:
It seems that the behavior for a MultiVariantWalker for managing sample collisions should be a tool specific behavior. I actually think you can hold off on adding the samples here in the code since it looks like everywhere else you use the headers you are manually merging the samples anyway in this branch so I would probably not open this can of worms. Alternatively, you could add a tool level override to the walker that that allows different headers to repeat samples.

Reply:
Aha, for some reason I missed MultiVariantWalker::getSamplesForVariants().  Problem solved.